### ISSUE TITLE
Schemas are parsed in spinal case instead of camel case when using path references

### ISSUE THREAD

**Issue by rombru** [openapi-processor/openapi-processor-spring#124]
Hello,

When using path reference,
```
  '/users/{userId}':
    $ref: ./apis/user.v1.yaml#/paths/~1users~1{userId}
```
The schemas used by the referenced path are parsed in spinal-case,
`public class user {`

For schemas whose name is in several parts, like UserSearch, the parser writes a hyphen (user-search) and stops because the syntax of java doesn't support that.

Because it's a bit complicated to exlain I've attached the full example
[openapi-bug.zip](https://github.com/openapi-processor/openapi-processor-spring/files/6104489/openapi-bug.zip)

**Comment by hauner** [comment]
Interesting, I didn't know you can navigate like this.  

I will take a look at which level this is going wrong. Thanks for the example. 

Reference:
https://stackoverflow.com/questions/58909124/how-to-use-ref-to-reference-a-path-from-another-openapi-file
https://stackoverflow.com/questions/44150758/swagger-2-0-multiple-path-objects-with-different-paths-but-same-request-and-res/44164413#44164413

**Comment by hauner** [comment]
The examples help a lot, Thanks.

The issue is that the processor tries to get schema names (titles) of $refs by extracting them from the $ref path. That fails for a lot of $refs in your example.

Looking at it now, I don't understand why it works at all ;-)

**Comment by hauner** [comment]
Fixed it. Both examples produce proper code now.  :-) 

That was a nasty one, but my original code wasn't so bad at all. The swagger parser is doing something very similar internally. :-)

Release will follow in the next days.

**Comment by rombru** [comment]
Nice ! So quick !

I have no doubts about the quality of the code. 
This is one of the reasons that pushes me to use it.

Thank you :smiley:

**Comment by hauner** [comment]
published 2021.3, fixing this one.
